src/Common/src/System/Net/Security/SecurityBuffer.cs

```diff
@@ -2,61 +2,89 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System.Diagnostics;
 using System.Runtime.InteropServices;
 using System.Security.Authentication.ExtendedProtection;
 
 namespace System.Net.Security
 {
-    internal class SecurityBuffer
+    // Until we have stackalloc Span<ReferenceType> support, these two
+    // structs allow us to do the equivalent of stackalloc SecurityBuffer[2]
+    // and stackalloc SecurityBuffer[3], with code like:
+    //     TwoSecurityBuffers tmp = default;
+    //     Span<SecurityBuffer> buffers = MemoryMarshal.CreateSpan<ref tmp._item0, 2);
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal ref struct TwoSecurityBuffers
     {
+        internal SecurityBuffer _item0;
+        private SecurityBuffer _item1;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    internal ref struct ThreeSecurityBuffers
+    {
+        internal SecurityBuffer _item0;
+        private SecurityBuffer _item1;
+        private SecurityBuffer _item2;
+    }
+
+    [StructLayout(LayoutKind.Auto)]
+    internal struct SecurityBuffer
```

Comment:
Since it's now Windows-only, can we actually move the file and namespace under Interop/Windows?

Reply:
I'll rename it to SecurityBuffer.Windows.cs.  As is though I don't think it belongs under Interop/Windows as it doesn't actually map 1:1 with any Windows types, to my knowledge.

Reply:
Ok.

It's not entirely obvious to me why some things are in Interop vs not; for example, SecuritySafeHandles is in Interop. But it doesn't really matter.

Reply:
> It's not entirely obvious to me why some things are in Interop vs not; for example, SecuritySafeHandles is in Interop

It probably shouldn't be there, though I'd need to look again at what it contains.  Our goal for the Interop folder was basically to be manually written equivalents of what you'd get if you had a tool that could spit out a perfect P/Invoke method and types for whatever you wanted to call, minimally opinionated.  So it would contain the signatures and types that map to the underlying APIs being invoked, no more, no less.  We've diverged a little from that in places, but I'd like to get back to that.